src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -55,7 +54,13 @@ object QueryService extends LazyLogging {
   def applyPrefix(value: String, prefixes: Map[String, String]): String =
     prefixes
       .filter(entry => value.startsWith(entry._2))
-      .map(entry => s"${entry._1}:" + value.substring(entry._2.length, value.length))
+      .map { entry =>
+        val prefix = s"${entry._1}:"
+        if (value.contains(prefix))
+          value.substring(value.lastIndexOf("/") + 1, value.length)
+        else
+          s"${entry._1}:" + value.substring(entry._2.length, value.length)
+      }
```

Comment:
This seems more complicated than I expect. Applying prefix should find the first entry `(prefix, expansion)` from the map where the value starts with the expansion. Then the expansion should be replaced with the prefix and a colon.

Reply:
MGI identifiers seem to be delimited by a ":".  We get curies like this: "MGI:MGI:894663" when not checking for `<prefix>:`.

Reply:
the following commit uses StringUtils.prefixIfMissing...same behavior, but cleaner